### DISCUSSION TITLE
Need help! I am stuck on this. Find the user who has many skills.

### DISCUSSION THREAD

**Discussion by axad1** [Asabeneh/30-Days-Of-JavaScript#406] [Q&A]
const txt = `{
    "Alex": {
        "email": "alex@alex.com",
        "skills": [
            "HTML",
            "CSS",
            "JavaScript"
        ],
        "age": 20,
        "isLoggedIn": false,
        "points": 30
    },
    "Asab": {
        "email": "asab@asab.com",
        "skills": [
            "HTML",
            "CSS",
            "JavaScript",
            "Redux",
            "MongoDB",
            "Express",
            "React",
            "Node"
        ],
        "age": 25,
        "isLoggedIn": false,
        "points": 50
    },
  }
  `

**Comment by dnthem** [comment]
I think what the question asks is to find the person who has the most skills. 

```js
let max = Number.MIN_SAFE_INTEGER;
let res = undefined;
for (const key in users)
{
     const user = users[key];
     if (user.skills.length > max)
    {
         max = user.skills.length;
         res = user;
     }
}
```

**Comment by palamarchukser** [comment]
Hey @AsadHusn , is this solution right for you?

```
const txt = { "Alex": { "email": "alex@alex.com", "skills": [ "HTML", "CSS", "JavaScript" ], "age": 20, "isLoggedIn": false, "points": 30 }, "Asab": { "email": "asab@asab.com", "skills": [ "HTML", "CSS", "JavaScript", "Redux", "MongoDB", "Express", "React", "Node" ], "age": 25, "isLoggedIn": false, "points": 50 }, };

let result = Object.keys(txt).reduce((a, b) => (txt[a].skills.length > txt[b].skills.length) ? a : b);

console.log(result) // For getting Username who has many skills
console.log(txt[result]) // For getting User object who has many skills
```

**Comment by vamsichintha** [comment]
const txt = { "Alex": { "email": "alex@alex.com", "skills": [ "HTML", "CSS", "JavaScript" ], "age": 20, "isLoggedIn": false, "points": 30 }, "Asab": { "email": "asab@asab.com", "skills": [ "HTML", "CSS", "JavaScript", "Redux", "MongoDB", "Express", "React", "Node" ], "age": 25, "isLoggedIn": false, "points": 50 }, }
if (txt.Alex.skills.length > txt.Asab.skills.length) {
  console.log("Alex has more skills (" + txt.Alex.skills + ")");  
} else {
  console.log("Asab has more skills (" + txt.Alex.skills + ")"); 
}

**Comment by JohnArbie** [comment]
function getMostskill() {
  let max = 1
  let maxSkill = {}
  for (const key in users) {
    const user = users[key]
    if(user.skills.length > max){
        maxSkill = user
        max = user.skills.length
    }
  }
  return maxSkill
}
